### PR TITLE
Added multiprocessing support and changed interface for shim functions

### PR DESCRIPTION
**Pull request by sternj**

### CODE REVIEW

File: scalene/replacement_pjoin.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import os
+from scalene.scalene_profiler import Scalene
+import sys
+import threading
+
+
+@Scalene.shim
+def replacement_pjoin(scalene):
+    def replacement_process_join(self, timeout: float = -1):
+        print("JOINING")
```
Comment (emeryberger) [inline scalene/replacement_pjoin.py:11]: left over debugging artifact

File: scalene/scalene_profiler.py [hunk 2]
```diff
@@ -494,6 +495,15 @@ def wrapper_profile(*args: Any, **kwargs: Any) -> Any:
         return wrapper_profile
 
     @staticmethod
+    def shim(func: Any) -> Any:
+        func(Scalene)
+        # Not sure why anyone would WANT to use this function, considering that it just takes in
```
Comment (emeryberger) [inline scalene/scalene_profiler.py:500]: This comment is uninformative - also, please add a docstring.

File: scalene/scalene_profiler.py [hunk 3]
```diff
@@ -1562,15 +1502,18 @@ def main() -> None:
                         exit_status = 0
                         profiler.start()
                         # Run the code being profiled.
+                        import sys as _sys
+                        _sys.argv = ['abcde']
```
Comment (emeryberger) [inline scalene/scalene_profiler.py:1506]: What's this for?

File: scalene/scalene_profiler.py [hunk 4]
```diff
@@ -1562,15 +1502,18 @@ def main() -> None:
                         exit_status = 0
                         profiler.start()
                         # Run the code being profiled.
+                        import sys as _sys
```
Comment (emeryberger) [inline scalene/scalene_profiler.py:1505]: This line needs an explanatory comment.

File: scalene/scalene_profiler.py [hunk 5]
```diff
@@ -1562,15 +1502,18 @@ def main() -> None:
...
                         try:
                             exec(code, the_globals, the_locals)
                         except SystemExit as se:
                             # Intercept sys.exit and propagate the error code.
                             exit_status = se.code
                         except BaseException:
-                            if Scalene.__debug:
-                                print(traceback.format_exc())  # for debugging only
-                            pass
+                            print(traceback.format_exc())  # for debugging only
```
Comment (emeryberger) [inline scalene/scalene_profiler.py:1515]: Mentioned for previous PR; should probably just do whatever Python does by default, which may not _exactly_ be this.

**Review by emeryberger — CHANGES_REQUESTED** [review]

File: scalene/scalene_profiler.py (hunk 5 above)
Comment (sternj) [reply scalene/scalene_profiler.py:1515]: Closed previous PR to consolidate-- this appears to be the default behavior, but I'll verify in CPython

File: scalene/scalene_profiler.py (hunk 3 above)
Comment (sternj) [reply scalene/scalene_profiler.py]: Debugging artifact from figuring out the behavior of `exec`

File: scalene/scalene_profiler.py (hunk 4 above)
Comment (sternj) [reply scalene/scalene_profiler.py]: Removed, as it was a debugging artifact

**Review by emeryberger — APPROVED** [review]

**Pull request merged**